pkg/resources/user_password_policy_attachment.go

```diff
@@ -0,0 +1,169 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var userPasswordPolicyAttachmentSchema = map[string]*schema.Schema{
+	"user_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "User name of the user you want to attach the password policy to",
+	},
+	"password_policy_database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Database name where the password policy is stored",
+	},
+	"password_policy_schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Schema name where the password policy is stored",
+	},
+	"password_policy_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Non-qualified name of the password policy",
+	},
+}
+
+func UserPasswordPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Specifies the password policy to use for the current account. To set the password policy of a different account, use a provider alias.",
+
+		Create: CreateUserPasswordPolicyAttachment,
+		Read:   ReadUserPasswordPolicyAttachment,
+		Delete: DeleteUserPasswordPolicyAttachment,
+
+		Schema: userPasswordPolicyAttachmentSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), helpers.IDDelimiter)
+				if len(parts) != 4 {
+					return nil, fmt.Errorf("id should be in the format 'database|schema|password_policy|user_name|roles', but I got '%s'", d.Id())
+				}
+				passwordPolicyDatabase := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[0])
```

Comment:
I don't get why these identifiers are created like that here. Is there a reason for customizing the importer instead of just using ImportStatePassthroughContext?

Reply:
Yeah, then the Terraform will just call the Read function (ReadUserPasswordPolicyAttachment in this case) to import state.

Reply:
The majority of the resources use the `ImportStatePassthroughContext` approach. All the ones using the complicated import are the grant ones, but they have their reasons: they are more complex resources, where more data needs to be provided to create a proper import. I prefer consistency: to have an `ImportStatePassthroughContext` here.

Reply:
No prob! Changed!